src/component.jl

```diff
@@ -0,0 +1,185 @@
+end
+
+function has_forecasts(component::InfrastructureSystemsType)
+    return !isnothing(_get_forecast_container(component))
+end
+
+function get_forecast_initial_times(
+                                    ::Type{T},
+                                    component::InfrastructureSystemsType,
+                                   ) where T <: Forecast
+    if !has_forecasts(component)
+        throw(ArgumentError("$(typeof(component)) does not have forecasts"))
+    end
+    return get_forecast_initial_times(forecast_external_to_internal(T),
+                                      _get_forecast_container(component))
+end
+
+function get_forecast_initial_times(
+                                    ::Type{T},
+                                    component::InfrastructureSystemsType,
+                                    label::AbstractString,
+                                   ) where T <: Forecast
+    if !has_forecasts(component)
+        throw(ArgumentError("$(typeof(component)) does not have forecasts"))
+    end
+    return get_forecast_initial_times(forecast_external_to_internal(T),
+                                      _get_forecast_container(component),
+                                      label)
+end
+
+function get_forecast_initial_times!(
+                                     initial_times::Set{Dates.DateTime},
+                                     component::InfrastructureSystemsType,
+                                    )
+    if !has_forecasts(component)
+        throw(ArgumentError("$(typeof(component)) does not have forecasts"))
+    end
+
+    get_forecast_initial_times!(initial_times, _get_forecast_container(component))
+end
+
+function get_forecast_keys(component::InfrastructureSystemsType)
+    return keys(_get_forecast_container(component).data)
+end
+
+function get_forecast_labels(
+                             ::Type{T},
+                             component::InfrastructureSystemsType,
+                             initial_time::Dates.DateTime,
+                            ) where T <: Forecast
+    return get_forecast_labels(forecast_external_to_internal(T), _get_forecast_container(component), initial_time)
+end
+
+function get_forecast_value(
+                            ::Type{T},
+                            component::InfrastructureSystemsType,
+                            initial_time::Dates.DateTime,
+                            label::AbstractString,
+                            index::Union{Int, Dates.DateTime},
+                           ) where T <: ForecastInternal
+    forecast = get_forecast(T, component, initial_time, label)
+    return get_timeseries(forecast)[index]
+end
+
+function get_time_series_uuids(component::InfrastructureSystemsType)
+    container = _get_forecast_container(component)
+
+    return [(get_time_series_uuid(container.data[key]), key.label)
+             for key in get_forecast_keys(component)]
+end
+
+function iterate_forecasts(component::InfrastructureSystemsType)
```

Comment:
We might want something like this that iterates over the forecasts with the same initial time, not the forecasts inside of a component. Unless is used internally for something else, I don't think we would need this feature.

Reply:
For debug purposes I commonly do something like this:
```
forecasts = collect(iterate_forecasts(component))
```